lib/resolvers/training-pil.js

```diff
@@ -0,0 +1,59 @@
+const { pick } = require('lodash');
+const moment = require('moment');
+const resolver = require('./base-resolver');
+const { generateLicenceNumber } = require('../utils');
+
+module.exports = ({ models }) => async ({ action, data, id }, transaction) => {
+  const { TrainingPil, Profile, Permission } = models;
+
+  if (action === 'create') {
+    let profile = await Profile.query(transaction).select('id', 'dob').findOne({ email: data.email });
+
+    if (profile) {
+      if (!profile.dob && data.dob) {
+        await profile.$query(transaction).patchAndFetch({ dob: data.dob });
```

Comment:
Is there a reason for `patchAndFetch` here rather than ordinary `patch` given the result is not being assigned?

Reply:
yes - editor autosuggest and laziness